codes/c/chapter_sorting/counting_sort.c

```diff
@@ -0,0 +1,81 @@
+/**
+ * File: counting_sort.c
+ * Created Time: 2023-03-20
+ * Author: Reanon (793584285@qq.com)
+ */
+#include "../include/include.h"
+
+/* 计数排序 */
+// 简单实现，无法用于排序对象
+void countingSortNaive(int nums[], int size) {
+    // 1. 统计数组最大元素 m
+    int m = 0;
+    for (int i = 0; i < size; i++) {
+        if (nums[i] > m) {
+            m = nums[i];
+        }
+    }
+    // 2. 统计各数字的出现次数
+    // counter[num] 代表 num 的出现次数
+    int *counter = malloc(sizeof(int) * size);
```

Comment:
Here the return value of malloc is `void *`, which needs to be converted to `int *`.

`int *counter =(int *) malloc(sizeof(int) * size);`

Reply:
It is unnecessary, as void * is automatically and safely promoted to any other pointer type in C.more in https://stackoverflow.com/questions/605845/do-i-cast-the-result-of-malloc

Reply:
> It is unnecessary, as void * is automatically and safely promoted to any other pointer type in C.more in https://stackoverflow.com/questions/605845/do-i-cast-the-result-of-malloc

ok, thanks @Reanon 

I took a look, there is no problem in C (sorry, I just used the g++ compiler)

Reply:
🤝